### PR TITLE
[scp] Don't set existing result as new result for action

### PR DESCRIPTION
**Pull request by NikitaZotov**
Related tasks

https://github.com/ostis-ai/sc-machine/pull/438

### CODE REVIEW

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp [hunk 1]
```diff
@@ -50,20 +50,13 @@ ScResult ASCPHandlingEventThatInitiatesAgentSCPProgram::DoProgram(ScElementaryEv
...
     }
   }
-  auto const & resIt = m_context.CreateIterator5(
-      action, ScType::ConstCommonArc, ScType::ConstNode, ScType::ConstPermPosArc, ScKeynodes::nrel_result);
-  if (resIt->Next())
-  {
-    ScAddr const & actionResult = resIt->Get(2);
-    action.SetResult(actionResult);
-  }
   return action.FinishSuccessfully();
```
Comment (kilativ-dotcom) [inline scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: something similar should be done with action.FinishSuccessfully
sometimes actions are finished successfully 2 times
also it is possible that scp-program finishes action unsuccessfully and then this agent generates arc from action_finished_successfully

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp [hunk 2]
```diff
@@ -13,6 +13,19 @@
...
+namespace
+{
+class SCPResult : public ScResult
+{
+public:
+  SCPResult(sc_result code)
+    : ScResult(code)
+  {
+  }
+};
```
Comment (kilativ-dotcom) [inline scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: move this class to separate cpp/hpp files

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp [hunk 3]
```diff
@@ -50,21 +63,28 @@ ScResult ASCPHandlingEventThatInitiatesAgentSCPProgram::DoProgram(ScElementaryEv
...
+    resultCode = SC_RESULT_NO;
+  else if (action.IsFinishedWithError())
+    resultCode = SC_RESULT_ERROR;
+  else
   {
-    ScAddr const & actionResult = resIt->Get(2);
-    action.SetResult(actionResult);
+    action.FinishSuccessfully();
+    resultCode = SC_RESULT_OK;
   }
```
Comment (kilativ-dotcom) [inline scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: there is one more case left
if action is finished with some status but isn't added to action_finished

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp (hunk 2 above)
Comment (NikitaZotov) [reply scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: No, it's only used in this file

Comment (kilativ-dotcom) [reply scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: This is not a valid reason
Most agents are only used in module file but they are placed in their own files

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp (hunk 3 above)
Comment (NikitaZotov) [reply scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: if action is finished with some status the one the first three cases is gonna pass

Comment (NikitaZotov) [reply scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: there is case when action is finished but it is not finished with some status

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp (hunk 2 above)
Comment (NikitaZotov) [reply scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: This is not java, in which all classes must have their own file. In C++ there is such a concept as encapsulation by hiding in a file. It is unlikely that SCPResult class will be used anywhere else

File: scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp (hunk 3 above)
Comment (kilativ-dotcom) [reply scp/scpHandlingEventThatInitiatesAgentSCPProgram.cpp]: I mean if action is finished successfully then this metaagent won't add it to action_finished, but all event subscriptions are subscribed to generation arc from action_finished

**Review by kilativ-dotcom — COMMENTED** [review]
left comment in a thread

**Review by kilativ-dotcom — APPROVED** [review]

**Pull request merged**